Model/Api/UpdateCartCommon.php

```diff
@@ -112,38 +112,33 @@ public function __construct(
      * @param  object $request
      * @return bool
      */
-    public function validateQuote( $parentQuoteId, $immutableQuoteId, $incrementId )
+    public function validateQuote($immutableQuoteId)
     {
-        if(empty($parentQuoteId)) {
-            $this->bugsnag->notifyError(
-                BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
-                'The cart.order_reference is not set or empty.'
-            );
-            $this->sendErrorResponse(
-                BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
-                'The cart reference is not found.',
-                404
-            );
-            return false;
-        }
-        
-        if(empty($incrementId)) {
-            $this->bugsnag->notifyError(
-                BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
-                'The cart.incrementId is not set or empty.'
-            );
-            $this->sendErrorResponse(
-                BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
-                'The cart incrementId is not found.',
-                404
-            );
-            return false;
-        }
-        
-        // check if the cart / quote exists and it is active
         try {
-            if (!$immutableQuoteId) {
-                $immutableQuoteId = $parentQuoteId;
+            // check the existence of child quote
+            $immutableQuote = $this->cartHelper->getQuoteById($immutableQuoteId);
+            if (!$immutableQuote) {
+                $this->sendErrorResponse(
+                    BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
+                    sprintf('The cart reference [%s] is not found.', $immutableQuoteId),
+                    404
+                );
+                return false;
+            }
+
+            $parentQuoteId = $immutableQuote->getBoltParentQuoteId();
+
+            if(empty($parentQuoteId)) {
+                $this->bugsnag->notifyError(
+                    BoltErrorResponse::ERR_INSUFFICIENT_INFORMATION,
+                    'Parent quote is not exist'
```

Comment:
@vitaliyreznikov Can you change this to "Parent quote does not exist" here and below?